### PR TITLE
Fix3180

### PR DESCRIPTION
**Pull request by gouttegd**
This pull request addresses issue #3180. 

The first commit implements the change discussed in #2831 and is the actual fix for issue #3180; the second commit fixes a unrelated bug that was revealed by the sample file provided in #3180, namely the fact that the parser could not properly handle features that “wrap” the end of the sequence.

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

**Comment by gouttegd** [comment]
Sorry about the failed CI, my Flake8 setup was not up to date and missed the Black violations. I’ll amend that shortly.

**Comment by peterjc** [comment]
Should also fix #3181 (origin wrapping locations)

File: Bio/SeqIO/SnapGeneIO.py [hunk 1]
```diff
@@ -115,7 +115,9 @@ def _parse_location(rangespec, strand, record):
     start = start - 1
     if start > end:
         # Range wrapping the end of the sequence
-        location = FeatureLocation(start, len(record), strand=strand) + FeatureLocation(0, end, strand=strand)
+        l1 = FeatureLocation(start, len(record), strand=strand)
+        l2 = FeatureLocation(0, end, strand=strand)
+        location = l1 + l2
```
Comment (peterjc) [inline Bio/SeqIO/SnapGeneIO.py:120]: To make black happy you could also have added more brackets, but this is clear too.

**Comment by peterjc** [comment]
Good to see @nicolazilio acknowledged in the git commit comment for providing the test case (via #3180)

**Comment by gouttegd** [comment]
Doh, I forgot that the `NEWS.rst` file lists the contributors for each release, I thought it was enough to have been listed there last year. Sorry about that, I’ll update that file now.

**Comment by peterjc** [comment]
Thank you!

**Pull request merged**